src/core/setup.cpp

```diff
@@ -1095,6 +1095,7 @@ cds_t* cdsSetup(nrs_t* nrs, setupAide options, occa::properties& kernelInfoBC)
   bool avmEnabled = false;
   {
     for(int is = 0; is < cds->NSfields; is++) {
+      if(!cds->compute[is]) continue;
```

Comment:
You'll need to add a similar check below on line 1109. Thank you for catching this!

Reply:
The one-line change does actually fix the issues I was seeing in Cardinal - but I just added the extra part on 1109, perhaps there are other setting combinations that we don't cover in Cardinal.

Thanks!